src/libhirte/common/config.c

```diff
@@ -0,0 +1,175 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "libhirte/hashmap/hashmap.h"
+#include "libhirte/ini/ini.h"
+
+#include "config.h"
+
+
+/*
+ * The application configuration object.
+ */
+struct hashmap *config;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+int option_compare(const void *a, const void *b, void *udata) { // NOLINT
+        const struct config_option *option_a = a;
+        const struct config_option *option_b = b;
+        return strcmp(option_a->name, option_b->name);
+}
+#pragma GCC diagnostic pop
+
+uint64_t option_hash(const void *item, uint64_t seed0, uint64_t seed1) {
+        const struct config_option *option_a = item;
+        return hashmap_sip(option_a->name, strlen(option_a->name), seed0, seed1);
+}
+
+int config_initialize() {
+        config = hashmap_new(sizeof(struct config_option), 0, 0, 0, option_hash, option_compare, NULL, NULL);
+        if (config == NULL) {
+                // error during hashmap initialization
+                return ENOMEM;
+        }
+        return 0;
+}
+
+void config_dispose() {
+        hashmap_free(config);
+}
+
+int config_load_complete_cofiguration(
+                const char *default_config_file,
+                const char *custom_config_file,
+                const char *custom_config_directory) {
+        int result = 0;
+
+        if (default_config_file != NULL) {
+                result = config_load_from_file(default_config_file);
+                if (result != 0) {
```

Comment:
Should starting `hirte` really fail if one of these are missing? For example, it will fail if `/etc/hirte.conf` is missing even if I'd override its config by a custom one (e.g. from the cli option). For production this might be ok, but for developing/testing its a bit harder to use. 
Maybe after loading config a kind of validation should be done, e.g. is config port really set?

Reply:
OK makes sense, users might delete /etc/hirte*.conf, but we should not fail if they are not there.

But the default config file should be in `/usr/share/{hirte|hirte-agent}/config/hirte-default.conf` so it's mandatory part of RPM. If user removes such file, we should fail.